analysis/annotations/annotations.go

```diff
@@ -0,0 +1,378 @@
+				comment.Text)
+		}
+	}
+	annotations.collectContentKeys()
+	return annotations, nil
+}
+
+func (pa ProgramAnnotations) loadPackageDocAnnotations(doc *ast.CommentGroup) {
+	// TODO: implementation
+}
+
+// loadFileAnnotations loads the annotation that are not tied to a specific ssa node. This includes:
+// - config annotations
+// - positional annotations
+func (pa ProgramAnnotations) loadFileAnnotations(logger *config.LogGroup, annotationContents []string, position string) {
+	if len(annotationContents) <= 1 {
+		logger.Warnf("ignoring argot annotation with no arguments at %s", position)
+		return
+	}
+	switch annotationContents[0] {
+	case ConfigTarget:
+		pa.loadConfigTargetAnnotation(logger, annotationContents, position)
+	}
+}
+
+// loadConfigTargetAnnotation loads a config annotation. Config annotations look like
+// "//argot:config tag SetOptions(option-name-1=value1,option-name-2=value2)" and are always linked to a specific problem
+// tag.
+func (pa ProgramAnnotations) loadConfigTargetAnnotation(logger *config.LogGroup, annotationContents []string, position string) {
+	if len(annotationContents) <= 2 {
+		logger.Warnf("argot:config expects a target tag and one or more SetOptions")
+		logger.Warnf("a comment is likely missing something at %s", position)
+	}
+	targetTag := annotationContents[1]
+	if _, present := pa.Configs[targetTag]; !present {
+		pa.Configs[targetTag] = map[string]string{}
+	}
+	idents := setOptionsRegex.FindStringSubmatch(strings.Join(annotationContents[2:], " "))
+	if len(idents) > 1 {
+		for _, arg := range parseAnnotationArgs(idents) {
+			// split something that should be option-name=option-value
+			argc := strings.Split(arg, "=")
+			if len(argc) < 2 {
+				logger.Warnf(
+					"argot:config comment ignored because SetOptions argument is not option-name=value at %s",
+					position)
+				return
+			}
+			if prevValue, isSet := pa.Configs[targetTag][argc[0]]; isSet {
+				logger.Warnf("argot:config option for %q already set to %q, ignoring annotation at %s",
+					argc[0], prevValue, position)
+			} else {
+				pa.Configs[targetTag][argc[0]] = argc[1]
+				logger.Debugf("set option %q to %q for problem tagged %q (annotation at %s)",
+					argc[0], argc[1], targetTag, position)
+			}
+		}
+	} else {
+		logger.Warnf("argot:config annotation encountered without matching SetOptions at %s", position)
+	}
+	return
+}
+
+func parseParamAnnotation(function *ssa.Function, annotationContent []string,
+	annotations *FunctionAnnotation, comment *ast.Comment) error {
+	paramName := annotationContent[1]
+	foundParam := false
+	for _, param := range function.Params {
+		if param.Name() == paramName {
+			foundParam = true
+			if _, ok := annotations.params[param]; !ok {
+				annotations.params[param] = []Annotation{}
+			}
+			paramAnnotations, err := parseAnnotationContent(annotationContent)
+			if err != nil {
+				return err
+			}
+			annotations.params[param] = append(annotations.params[param], paramAnnotations...)
```

Comment:
nit: could `break` out of the loop here since the parameter has been found

Reply:
Yes, actually should refactor this to just return